File: backend/app/schemas/credential.py
from pydantic import BaseModel, Field, validator
from typing import Optional, Literal
from datetime import datetime


class CredentialBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    provider: Literal["openai", "anthropic", "claude_code"]
    api_url: Optional[str] = None


class CredentialCreate(CredentialBase):
    api_key: str = Field(..., min_length=1)

    @validator('api_url')
    def validate_api_url(cls, v, values):
        if v and not v.startswith(('http://', 'https://')):
            raise ValueError('API URL must start with http:// or https://')
        return v


class CredentialUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    api_key: Optional[str] = Field(None, min_length=1)
    api_url: Optional[str] = None
    is_active: Optional[bool] = None

    @validator('api_url')
    def validate_api_url(cls, v):
        if v and not v.startswith(('http://', 'https://')):
            raise ValueError('API URL must start with http:// or https://')
        return v


class CredentialResponse(CredentialBase):
    id: str
    user_id: str
    is_active: bool
    is_validated: bool
    validation_error: Optional[str]
    created_at: datetime
    updated_at: datetime
    api_key_masked: str  # 只显示前几位和后几位

    class Config:
        from_attributes = True


class CredentialValidate(BaseModel):
    """验证凭证响应"""
    is_valid: bool
    error_message: Optional[str] = None
    available_models: Optional[list[str]] = None

File: frontend/src/types/credential.ts
export type Provider = 'openai' | 'anthropic' | 'claude_code' | 'gemini' | 'ernie' | 'qwen' | 'azure_openai';

export interface ModelValidationResult {
  is_valid: boolean;
  error?: string;
  validated_at: string;
}

export interface Credential {
  id: string;
  user_id: string;
  name: string;
  provider: Provider;
  api_url?: string;
  custom_models: string[];
  model_validation_results?: Record<string, ModelValidationResult>;
  is_active: boolean;
  is_validated: boolean;
  validation_error?: string;
  created_at: string;
  updated_at: string;
  api_key_masked: string;
}

export interface CredentialCreate {
  name: string;
  provider: Provider;
  api_key: string;
  api_url?: string;
  custom_models?: string[];
}

export interface CredentialUpdate {
  name?: string;
  api_key?: string;
  api_url?: string;
  custom_models?: string[];
  is_active?: boolean;
}

export interface CredentialValidation {
  is_valid: boolean;
  error_message?: string;
  available_models?: string[];
  failed_models?: string[];
  model_validation_results?: Record<string, ModelValidationResult>;
  total_models_tested?: number;
  validation_summary?: string;
}

export interface CredentialState {
  credentials: Credential[];
  selectedCredential: Credential | null;
  isLoading: boolean;
  validationStatus: Record<string, boolean>;
}